### ISSUE TITLE
Attention_RNN: Nan Loss during training

### ISSUE THREAD

**Issue by miraodasilva** [tensorflow/magenta#1202]
Hello.

I have tried training a basic_rnn with  my own dataset and it worked well, producing a usable checkpoint. However, when I try to use an attention_rnn instead, by adding the "attn_length=40" to the hparams, I get the error "NaN loss during training.". I have tried chaning the attn_length to other values such as 10 or 20 and I still get this error. Also, I made sure to create the dataset using the "attention_rnn" parameter, so that should not be an issue. 

Anyone have any similar issues? 

Here are the commands I used:
"
convert_dir_to_note_sequences \
  --input_dir=$INPUT_DIRECTORY \
  --output_file=$SEQUENCES_TFRECORD \
  --recursive

melody_rnn_create_dataset --config="attention_rnn" --input=".../mono_notesequences.tfrecord" --output_dir="..." --eval_ratio="0.10"

python ${MODEL}/melody_rnn_train.py --config=attention_rnn --run_dir=${OUTPUT} --sequence_example_file=${INPUT}/attention_rnn/training_melodies.tfrecord --hparams="batch_size=128,rnn_layer_sizes=[512,512],attn_length=40" --num_training_steps=20000
"
Thank you very much!

**Comment by miraodasilva** [comment]
It turns out that using smaller layer_sizes such as [128,128] solves the issue. Does anyone have any idea why this is the case? Perhaps lowering the learning rate can allow me to use larger layer_sizes? 

Thanks.

**Comment by Mahsarm** [comment]
I'm training a data set with max seq length of 150 for thedata set of 2000/300/300 sketches and I am getting nan cost after about 30k steps, anybody has any idea why is that?

**Comment by miraodasilva** [comment]
I believe Nan loss is caused, in this case, by exploding/vanishing gradients. In order to fix this you can decrease the size of your layers, as I did, or try decreasing the clip_norm in order to prevent exploding gradients. Let me know if any of these options help.
